Common/src/main/java/com/techarium/techarium/client/screen/MachineCoreScreen.java

```diff
@@ -65,51 +74,50 @@ protected void init() {
 	 */
 	private void initButtons(String filter) {
 		this.startIndex = 0;
-		this.buttons.forEach(this::removeWidget);
 		this.buttons.clear();
-		Optional<ResourceLocation> selected = this.menu.selectedMultiblock();
-		ArrayList<MultiBlockButton> list = new ArrayList<>();
-		for (int i = 0; i < this.ids.size(); i++) {
-			ResourceLocation id = this.ids.get(i);
-			MutableComponent name = Utils.translatableComponent(id.toLanguageKey("multiblock"));
-			if (filter.isEmpty() || name.getString().toLowerCase().startsWith(filter.toLowerCase())) {
-				MultiBlockButton multiBlockButton = new MultiBlockButton(i, this.leftPos + IDS_X, 0, 100, 14, name, button -> {
-					this.buttons.forEach(element -> element.setSelected(false));
-					this.menu.setMultiBlock(id);
-					((MultiBlockButton) button).setSelected(true);
+
+		List<MultiBlockButton> list = new ArrayList<>();
+		for (var entry : multiblocks) {
+			MutableComponent name = Utils.translatableComponent(entry.getKey().location().toLanguageKey("multiblock"));
+			if (name.getString().toLowerCase(Locale.ENGLISH).contains(filter.toLowerCase(Locale.ENGLISH))) {
+				MultiBlockButton multiBlockButton = new MultiBlockButton(menu.getMachineCore(), entry.getValue(), this.leftPos + IDS_X, 0, 100, 14, name, button -> {
+					this.menu.getMachineCore().setMultiblock(entry.getValue());
 				});
-				if (selected.isPresent()) {
-					if (id.equals(selected.get())) {
-						multiBlockButton.setSelected(true);
-					}
-				}
+
 				list.add(multiBlockButton);
 			}
 		}
+
 		// sort the buttons by the translated name of the multiblocks
-		Collections.sort(list);
+		list = list.stream().sorted(Comparator.comparing(button -> button.getMessage().getString())).toList();
```

Comment:
would you please explain to me why the comparator should be here and not it the button class itself ?

Reply:
I don't think it makes sense for a button to be explicitly comparable.